### PR TITLE
 A few IntelliJ inspection-driven cleanups

### PR DESCRIPTION
**Pull request by stevencrockett**
## Overview

First time contributing to junit. When I first look at any project I usually try and do some simple cleanups as I get used to the code. Most of this was driven by intellij, but I made some other changes where I saw fit.

I'm hoping the commits are self-explanatory. Since they are small I just put them in one PR. There are other cleanups to be done but I'll stick with this for now to figure out the process. Let me know if there are changes you want to see.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

**Comment by sbrannen** [comment]
Tentatively slated for _review_ before 5.4 GA.

**Comment by sbrannen** [comment]
_in progress_

File: junit-platform-commons/src/main/java/org/junit/platform/commons/support/ReflectionSupport.java [hunk 1]
```diff
@@ -63,7 +63,6 @@ private ReflectionSupport() {
 	 */
 	@API(status = DEPRECATED, since = "1.4")
 	@Deprecated
-	@SuppressWarnings("deprecation")
```
Comment (sbrannen) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/support/ReflectionSupport.java]: I'm leaving this one in place for Eclipse users.

File: platform-tests/src/test/java/org/junit/platform/launcher/core/LauncherDiscoveryRequestBuilderTests.java [hunk 2]
```diff
@@ -199,36 +198,31 @@ void engineFiltersAreStoredInDiscoveryRequest() {
...
+					.build();
 			// @formatter:on
 
-			List<String> filterStrings = discoveryRequest.getFiltersByType(DiscoveryFilter.class).stream().map(
-				DiscoveryFilter::toString).collect(toList());
-			assertThat(filterStrings).hasSize(2);
-			assertThat(filterStrings).contains("filter1", "filter2");
+			var filters = discoveryRequest.getFiltersByType(DiscoveryFilter.class);
+			assertThat(filters).containsOnly(filter1, filter2);
 		}
```
Comment (sbrannen) [inline platform-tests/src/test/java/org/junit/platform/launcher/core/LauncherDiscoveryRequestBuilderTests.java]: Those test methods in `LauncherDiscoveryRequestBuilderTests` are much cleaner now. Thanks.

File: platform-tooling-support-tests/src/main/java/platform/tooling/support/Helper.java [hunk 3]
```diff
@@ -115,17 +115,8 @@ static JarFile createJarFile(String module) {
...
-		}
-		// Try to inspect Maven Toolchains configuration file...
-		var jdkHome = getJdkHomeFromMavenToolchains(version);
-		if (jdkHome.isPresent()) {
-			return jdkHome;
-		}
-		// Still here? Return an empty optional.
-		return Optional.empty();
+		var homePath = sources.stream().map(Supplier::get).filter(Objects::nonNull).findFirst().map(Path::of);
+		return homePath.or(() -> getJdkHomeFromMavenToolchains(version));
```
Comment (sbrannen) [inline platform-tooling-support-tests/src/main/java/platform/tooling/support/Helper.java]: I'm reverting this change.

Rationale: the original author of this code chose to split up the steps and add comments in order to simplify understanding of the steps involved.

**Comment by sbrannen** [comment]
Thanks for the contribution!

I merged most of the work in the last 4 commits pushed to `master` (starting with c56624fedcbeec7a0adff9229b588a2227bcbb7e).

File: platform-tooling-support-tests/src/main/java/platform/tooling/support/Helper.java (hunk 3 above)
Comment (stevencrockett) [reply platform-tooling-support-tests/src/main/java/platform/tooling/support/Helper.java]: I found it pretty unidiomatic use of the optional api, so to me at least this procedural style with all the split out steps made it less readable. As for the comments - in this case I didn't think the comments were adding anything - they are telling the same story as the code, which is a good place to be in. I can see where you're coming from though.

**Comment by stevencrockett** [comment]
Cool! Thanks for the review. :)

**Pull request closed without merging**